src/Microsoft.SqlTools.ServiceLayer/LanguageServices/PeekDefinition.cs

```diff
@@ -37,9 +39,20 @@ private Database Database
             {
                 if (this.connectionInfo.SqlConnection != null)
                 {
-                    Server server = new Server(this.connectionInfo.SqlConnection.DataSource);
-                    return server.Databases[this.connectionInfo.SqlConnection.Database];
-
+                    try
+                    {
+                        // Get server object from connection
+                        string connectionString = ConnectionService.BuildConnectionString(this.connectionInfo.ConnectionDetails);
+                        SqlConnection sqlConn = new SqlConnection(connectionString);                    
+                        sqlConn.Open();
```

Comment:
Note to others: issue https://github.com/Microsoft/vscode-mssql/issues/468 covers the need to fix this and cache / reuse the connection since Intellisense already has the required info